src/bobbit/modules/duckhunt.py

```diff
@@ -92,12 +98,19 @@ async def ducks(bot, message, command):
             return message.copy(body=f"No ducks are scheduled to visit {channel}.")
         if not Ducks[channel]:
             return message.copy(body=f"There are currently no ducks in {channel}.")
+        
+        # Check the cooldown
+        if Cooldowns[channel].get(nick, 0) > current_time:
```

Comment:
Might need to make this: `Cooldowns.get(channel, {}).get(nick, 0)` in case the channel is not yet in `Cooldowns`

Reply:
GP. will fix.